bin/hdfcoinc/pycbc_page_foreground

```diff
@@ -31,32 +59,62 @@ if args.output_file.endswith('.xml') or args.output_file.endswith('.xml.gz'):
         err_msg = "If creating xml files must provide the single detector "
         err_msg += "trigger lists with --single-detector-triggers."
         raise ValueError(err_msg)
-    
+
 fortrigs = hdf.ForegroundTriggers(args.trigger_file, args.bank_file,
-                       sngl_files=args.single_detector_triggers,
-                       n_loudest=args.num_coincs_to_write)
+                                  sngl_files=args.single_detector_triggers,
+                                  n_loudest=args.num_coincs_to_write)
 
 if args.output_file.endswith('.html'):
+    # If hierarchical removals were done and the user wants to look at a
+    # particular stage of hierarchical removals (h_num_rm), remove that
+    # number of loudest triggers by stat from the list of triggers.
+    if h_num_rm >= 0:
+        fortrigs.ignore_x_loudest(h_num_rm)
+
     ifar = fortrigs.get_coincfile_array('ifar')
-    ifar_exc = fortrigs.get_coincfile_array('ifar_exc') 
+    ifar_exc = fortrigs.get_coincfile_array('ifar_exc')
     fap = fortrigs.get_coincfile_array('fap')
     fap_exc = fortrigs.get_coincfile_array('fap_exc')
     stat = fortrigs.get_coincfile_array('stat')
     time1 = fortrigs.get_coincfile_array('time1')
     time2 = fortrigs.get_coincfile_array('time2')
-
     mass1 = fortrigs.get_bankfile_array('mass1')
     mass2 = fortrigs.get_bankfile_array('mass2')
     spin1z = fortrigs.get_bankfile_array('spin1z')
     spin2z = fortrigs.get_bankfile_array('spin2z')
 
+    # Write the correct false alarm rates and other significance values
+    # to the table after h_num_rm hierarchical removals
+    if h_num_rm >= 0:
```

Comment:
Steven, this seems dangerous. Why aren't you simply using the trigger_id columns that are written out? This is what the class underyling is doing. As Ian says, the best thing would be to move things to the underlying class. You an add another argument that tells it what key group to use, etc. 

Just save out the same information from pycbc_coinc_statmap like you did for the background groups. https://github.com/ligo-cbc/pycbc/blob/master/bin/hdfcoinc/pycbc_coinc_statmap#L308